index.php

```diff
@@ -167,12 +181,32 @@ function base64_url_decode($string)
     $redirect_uri = filter_input(INPUT_POST, 'redirect_uri', FILTER_VALIDATE_URL);
     $client_id = filter_input(INPUT_POST, 'client_id', FILTER_VALIDATE_URL);
 
+    // Scan through the existing users then
     // Exit if there are errors in the client supplied data.
-    if (!(is_string($code)
-        && is_string($redirect_uri)
-        && is_string($client_id)
-        && verify_signed_code(APP_KEY, USER_URL . $redirect_uri . $client_id, $code))
-    ) {
+    $user_files = scandir($configdir);
+    $user_verified = false;
+    foreach ($user_files as $user_file) {
+        // make sure we are only looking at proper config files
+        // when single-user, this will only ever be one file
+        // when multi-user, this will be the user files
+        if (preg_match('/^config(_\.+)?\.php$/', $user_file) !== 1) {
```

Comment:
extra character in regex
```suggestion
        if (preg_match('/^config(_.+)?\.php$/', $user_file) !== 1) {
```